File: app/src/main/java/social/entourage/android/EntourageApplicationModule.java
package social.entourage.android;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
final class EntourageApplicationModule {

    private final EntourageApplication app;

    public EntourageApplicationModule(final EntourageApplication app) {
        this.app = app;
    }

    @Provides
    @Singleton
    public Application providesApplication() {
        return app;
    }
}

File: app/src/main/java/social/entourage/android/EntourageApplicationModule.kt
package social.entourage.android

import android.app.Application
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
internal class EntourageApplicationModule(private val app: EntourageApplication) {
    @Provides
    @Singleton
    fun providesApplication(): Application {
        return app
    }

}